src/Engine/include/OpenLoco/Engine/World.hpp

```diff
@@ -88,4 +89,14 @@ namespace OpenLoco::World
     {
         return Pos2{ static_cast<coord_t>(coords.x * kTileSize), static_cast<coord_t>(coords.y * kTileSize) };
     }
+
+    constexpr coord_t clampCoord(coord_t coord)
```

Comment:
I'm not sure these are quite correct. We have basically two possible range limits. drawable and normal. drawable is 1 tile less than max/min and normal is max/min. This looks to be a bit of a mix of both. We should also probably define a drawable max min values as they are used a few times now.

Reply:
You are right but I don't think this is used anywhere in the drawing code at the moment, I also mentioned the oddity to @AaronVanGeffen on Discord that we should probably have separate constants for the world space as the +1, -1 offset seems quite odd to have for when we should rather have proper constants.

Reply:
Any suggestions for what those constants should look like?

Reply:
I think the code is fine as is, accessing tiles at 0,0 or the max is an invalid memory read. The drawing code interprets coordinates out of that range as void. We can refine the constants at a later point to better separate whats used for rendering and whats used for accessing the tile memory. @duncanspumpkin what are your thoughts?

Reply:
there are also tile manager functions that use the drawableCoords validator so i think we do need this to be correct for the various setups.

Reply:
Not sure I understand what you mean, when it comes to getting a Tile then it must be within the `1, 1` and `kkMapWidth, kMapColumns`, I don't see any code modified that isn't about getting memory for the tiles currently.

Reply:
I'm concerned about minor divergences. At present for example look at the audio change this was previously ranged to max less< kMapColumns now it is to 1-> kMapColumns-1. The upper range is the same as it's gone from < to <=. The lower range has truncated by 1. <0 we can ignore as it would crash before. And that's just for the tile clamp. For the world clamp it's also had its lower bound truncated from 0 to 1. Which doesn't match a drawable coord (32) or a world coord (0). The upper bound is still ultimately the same though as before.

So either the clamps should be 0->Columns-1 or they should be 1->Columns-2 and for world coords 0->Width-1 or 32->Width-33.

Reply:
Hm perhaps we should have two distinct functions then, getWorldRange and getDrawableRange and hide the clamping functions. Given the new functions the code shouldn't have to manually clamp coordinates anymore.

Reply:
is this resolved?

Reply:
I believe I've now adjusted the clamps as suggested.